core-build/gulp-core-build-typescript/src/RSCTask.ts

```diff
@@ -6,62 +6,46 @@ import * as resolve from 'resolve';
 
 import {
   JsonFile,
-  IPackageJson
+  IPackageJson,
+  FileSystem,
+  PackageJsonLookup
 } from '@microsoft/node-core-library';
 import { GulpTask } from '@microsoft/gulp-core-build';
 import * as RushStackCompiler from '@microsoft/rush-stack-compiler';
 import { GCBTerminalProvider } from './GCBTerminalProvider';
 
 export interface IRSCTaskConfig extends Object {
   buildDirectory: string;
+}
 
-  /**
-   * This is the name of the rush-stack-compiler package, or the name of the package that
-   * extends the configuration of rush-stack-compiler. This defaults to "\@microsoft/rush-stack-compiler"
-   */
-  rushStackCompilerPackageName?: string;
+interface ITsconfig {
+  extends?: string;
 }
 
 export abstract class RSCTask<TTaskConfig extends IRSCTaskConfig> extends GulpTask<TTaskConfig> {
+  private static _rushStackCompilerPackagePathCache: { [buildFolder: string]: string } = {};
+  private static __packageJsonLookup: PackageJsonLookup | undefined; // tslint:disable-line:variable-name
```

Comment:
Is it reasonable for these to be static?  It would imply that package.json and tsconfig.json files never change during the lifetime of the **rush-stack** process

Reply:
I think it's reasonable to assume that those files won't change during a build.

---
In reply to: [229790059](https://github.com/Microsoft/web-build-tools/pull/919#discussion_r229790059) [](ancestors = 229790059)